com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkPrefabHandler.cs

```diff
@@ -40,7 +40,8 @@ public interface INetworkPrefabInstanceHandler
         /// The most common approach is to make the <see cref="NetworkObject"/> inactive by calling <see cref="GameObject.SetActive(bool)"/>.
         /// </summary>
         /// <param name="networkObject">The <see cref="NetworkObject"/> being destroyed</param>
-        void Destroy(NetworkObject networkObject);
+        /// <returns>(true) destroy the <see cref="GameObject"/> (false) do not destroy the <see cref="GameObject"/> </returns>
+        bool Destroy(NetworkObject networkObject);
```

Comment:
This is specifically for when a registered INetworkPrefabInstanceHandler implementation wants a NetworkObject to be fully destroyed it can return "true" and it will be destroyed fully (i.e. GameObject and all).  Since clients cannot destroy NetworkObjects, this provides the mechanism for a client to have control over when a NetworkObject should be destroyed upon having received a Despawn message from the server ( see NetworkSpawnManager.OnDespawnObject).  
One example would be if a user wants all NetworkObjects spawned by a Network Prefab Handler aware pooled spawn generator to be destroyed or not destroyed when an additive scene is unloaded but the NetworkObjects reside in the currently active scene (i.e. single mode loaded scene).